src/core/conn.h

```diff
@@ -27,11 +27,8 @@ class CFunctionBlock;
 
 class CConnectionPoint {
   public:
-    CFunctionBlock *mFB;
-    TPortId mPortId;
-
-    CConnectionPoint(CFunctionBlock *paFB, TPortId paPortId) :
-      mFB(paFB), mPortId(paPortId) {
+    CConnectionPoint(CFunctionBlock &paFB, const TPortId paPortId) :
+      mFB(&paFB), mPortId(paPortId) {
     }
 
     CConnectionPoint() = default;
```

Comment:
is this constructor needed? since now the members cannot be set without the other constructor

Reply:
the ringbuffer only use the assign operator to store the data in the buffer, so if we had the assign operator for this class it should work (this analysis is done only from checking the code, didn't try compiling it yet without the constructor)

Reply:
ok, I tried doing it locally and I see the problem now regarding `std::array`. Forget about this comment

Reply:
I was offline and about to write that I tried a few things. But the problem is that references are not assignable.  STL has the `std::reference_wrapper` for this task but in the end it is doing the same thing that we are doing here. And as this is in addition a very performance sensitive class I don't think it is worth the effort.